tests/test_covariance_clusters.py

```diff
@@ -41,33 +42,49 @@ def mock_sacc():
     # /blob/master/Full%20covariance%20N_N%20part%20vfinal.ipynb
     # As reference.
     s = sacc.Sacc()
-    s.metadata["nbins_cluster_redshift"] = 18
-    s.metadata["nbins_cluster_richness"] = 3
-    s.metadata["min_mass"] = 1e13
-
-    # This isnt how tracers actually look, but sort of
-    # hacks the class to work without building
-    # an entire sacc file for this test.
-    s.add_tracer(
-        "Misc",
-        "clusters_0_0",
-        metadata={
-            "Mproxy_name": "richness",
-            "Mproxy_min": 10,
-            "Mproxy_max": 100,
-            "z_name": "redshift",
-            "z_min": 0.3,
-            "z_max": 1.2,
-        },
-    )
+    z_min, z_max = 0.3, 1.2
+    richness_min, richness_max = np.log10(10), np.log10(100)
+
+    z_tracers = []
+    z_edges = np.linspace(z_min, z_max, 19)
+    for i, zbin in enumerate(zip(z_edges[:-1], z_edges[1:])):
+        bin_z_label = f"bin_z_{i}"
+        s.add_tracer("bin_z", bin_z_label, zbin[0], zbin[1])
+        z_tracers.append(bin_z_label)
+
+    richness_tracers = []
+    richness_edges = np.linspace(richness_min, richness_max, 4)
+    for i, richness_bin in enumerate(
+        zip(richness_edges[:-1], richness_edges[1:])
+    ):
+        bin_richness_label = f"bin_rich_{i}"
+        s.add_tracer(
+            "bin_richness",
+            bin_richness_label,
+            richness_bin[0],
+            richness_bin[1],
+        )
+        richness_tracers.append(bin_richness_label)
+
+    tracer_combos = list(itertools.product(z_tracers, richness_tracers))
+    cluster_counts = np.linspace(1, 1e4, len(tracer_combos))
+
+    counts_and_edges = zip(cluster_counts.flatten(), tracer_combos)
+
+    for counts, (z_tracers, richness_tracers) in counts_and_edges:
+        s.add_data_point(
+            sacc.standard_types.cluster_counts,
+            (bin_z_label, bin_richness_label),
+            int(counts),
+        )
 
     return s
 
 
 @pytest.fixture
 def mock_covariance_gauss(mock_sacc, mock_cosmo):
     cc_cov = ClusterCountsGaussian(INPUT_YML)
-    cc_cov.load_from_sacc(mock_sacc)
```

Comment:
Hmm, if there is a sacc file in the yml file, why isn't it using it? Also, I'd suggest generating one on the fly to avoid having to keep track of extra files, but not very important.

Reply:
The only reference I have is Nelson's original notebooks, which had very specific input data that is nothing like sacc.  I'll look into trying to create a sacc file which has the same contents, but last time it seemed difficult.

Reply:
Got it working